### DISCUSSION TITLE
High proportion of '__unknown__' values in TagProf report

### DISCUSSION THREAD

**Discussion by oskarpearson** [test-prof/test-prof#286] [Q&A] [answered]
Hi All

I'm busy trialling test-prof on our rspec-based system. When using `TAG_PROF=type` things seem to work mostly as one would expect.

In the output there are the standard categories I would expect - things like controller/system/worker etc.

However, quite a large proportion (9th on our list) is categorised as `__unknown__`.

From a look at pull requests and the code, it seems like this should only be happening with mintest? https://github.com/test-prof/test-prof/blob/bd7027e1cd58d58cd4f129f5c41012917a6a18c4/docs/profilers/tag_prof.md?plain=1#L56 Since we're using rspec, this seems a bit confusing.

- Does this `unknown` category indicate that some of our rspec tests don't have the expected metadata? (ie `metadata[:type]`)
- Or does this `unknown` category mean that there's something happening outside of the rspec framework (and invisible to TagProf reporting) that is slowing things down?

**Comment by palkan** [comment, marked as answer]
Hey!

> Does this unknown category indicate that some of our rspec tests don't have the expected metadata? (ie metadata[:type])

This.

Do you have the `config.infer_spec_type_from_location!` in your `rails_helper.rb`? Also, have you added the `defined_derived_metadata` configuration ([docs](https://test-prof.evilmartians.io/profilers/tag_prof?id=pro-tip-more-types))?

**Reply by oskarpearson** [reply]
Thanks - that definitely helps.

We already have an existing `infer_spec_type_from_location` setup in our code that's a bit unusual. This is because we use Packwerk, and our specs have been split out into per-component folders,  with model/controller/etc tests in each component.

So your answer indicates that we're missing some classification in the existing infer_spec_type_from_location setup. Thank you
